docs/shared/packages/playwright/playwright-plugin.md

```diff
@@ -0,0 +1,209 @@
+
+### Specifying a Base Url
+
+The `baseURL` property within the Playwright configuration can control where the tests visit by default.
+
+```ts
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  // Rest of your config...
+
+  // Run your local dev server before starting the tests
+  webServer: {
+    command: 'npx serve <your-app-name>',
+    url: 'http://localhost:4200',
+    reuseExistingServer: !process.env.CI,
+  },
+  use: {
+    baseURL: 'http://localhost:4200', // url playwright visits with `await page.goto('/')`;
+  },
+});
+```
+
+In order to set different `baseURL` values for different environments you can pass them via the [environment variables and nx configurations](/recipes/tips-n-tricks/define-environment-variables) or optionally via setting them per the environment they are needed in such as `CI`
+
+```ts
+import { defineConfig } from '@playwright/test';
+
+const baseUrl =
+  process.env.BASE_URL ?? process.env.CI
+    ? 'https://some-staging-url.example.com'
+    : 'http://localhost:4200';
+
+export default defineConfig({
+  // Rest of your config...
+
+  // Run your local dev server before starting the tests
+  webServer: {
+    command: 'npx serve <your-app-name>',
+    url: baseUrl,
+    reuseExistingServer: !process.env.CI,
+  },
+  use: {
+    baseURL: baseUrl, // url playwright visits with `await page.goto('/')`;
+  },
+});
+```
+
+By default Nx, provides a `nxE2EPreset` options in the Playwright configuration.
+
+```ts
+import { defineConfig } from '@playwright/test';
+import { nxE2EPreset } from '@nx/playwright/preset';
+import { workspaceRoot } from '@nx/devkit';
+
+// For CI, you may want to set BASE_URL to the deployed application.
+const baseURL = process.env['BASE_URL'] || 'http://localhost:4200';
+
+/**
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// require('dotenv').config();
+
+/**
+ * See https://playwright.dev/docs/test-configuration.
+ */
+export default defineConfig({
+  ...nxE2EPreset(__filename, { testDir: './e2e' }),
+  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+  use: {
+    baseURL,
+    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
+    trace: 'on-first-retry',
+  },
+  /* Run your local dev server before starting the tests */
+  webServer: {
+    command: 'npx nx serve <your-app-name>',
+    url: baseURL,
+    reuseExistingServer: !process.env.CI,
+    cwd: workspaceRoot,
+  },
+});
+```
+
+This preset sets up the `outputDir` and [HTML reporter](https://playwright.dev/docs/test-reporters#html-reporter) to output in `dist/.playwright/<path-to-project-root>`, setups chromium, firefox, and webkit browsers. If you want to use mobile and/or branded browsers you can pass those options into the preset function
```

Comment:
`setups chromium, firefox, and webkit browsers`. I think you're missing some words here.  Not sure what this sentence should be.